### PR TITLE
Implemented: Add support to update current timezone from setting page(#29bdmvr)

### PR DESCRIPTION
**Pull request by Mayank909**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/commerce-hub#contribution-guideline)

https://user-images.githubusercontent.com/44576058/167127588-d3abd0f7-b8d8-4181-aaf8-cc348cda5eb8.mp4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "@vue/cli-plugin-router": "~4.5.0",
     "@vue/cli-plugin-typescript": "~4.5.0",
     "@vue/cli-plugin-unit-jest": "~4.5.0",
-    "@vue/cli-service": "~4.5.0",
+    "@vue/cli-service": "4.5.17",
```
Comment (ymaheshwari1) [inline package.json]: We will not make any changes to package.json file until we add a new package or we update version for some package. Please revert this change as I think this version upgrade is not needed and also revert the change for package-lock.json file

File: src/views/TimezoneModal.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+     <ion-toolbar>
+        <ion-buttons slot="start">
+        <ion-button @click="closeModal"> 
+          <ion-icon :icon="close" />
+        </ion-button>
+      </ion-buttons>
+     <ion-title>{{ $t("Select time zone") }}</ion-title>
+    </ion-toolbar>
+    <ion-toolbar>
+        <ion-searchbar @ionFocus="selectSearchBarText($event)" :placeholder="$t('Search time zones')"  v-model="queryString" @ionInput="findTimeZone()" />
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue]: Improve indentation.

File: src/views/TimezoneModal.vue [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    <!-- Empty state -->
+    <div class="empty-state" v-if="filteredTimeZones.length === 0">
+      <p>{{ $t("No time zone found")}}</p>
+    </div>
+
+    <!-- Timezones -->
+    <div v-else>
+      <ion-list>
+        <ion-radio-group value="rd" v-model="timeZoneId">
+          <ion-item  v-bind:key="timeZone.id" v-for="timeZone in filteredTimeZones">
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue]: When using `:` we don't need to use `v-bind`, `:` is shorthand for `v-bind`.

File: src/views/TimezoneModal.vue [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+      );
+      this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
+        return regularExp.test(timeZone.id) || regularExp.test(timeZone.label);
+      });
+    },
+    async getAvailableTimeZones() {
+      UserService.getAvailableTimeZones().then((resp: any) => {
+        if (resp.status === 200 && !hasError(resp)) {
+          this.timeZones = resp.data.filter((timeZone: any) => {
+              return moment.tz.zone(timeZone.id);
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue]: Improve indentation.

File: src/views/TimezoneModal.vue [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        "i"
+      );
+      this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
+        return regularExp.test(timeZone.id) || regularExp.test(timeZone.label);
+      });
+    },
+    async getAvailableTimeZones() {
+      UserService.getAvailableTimeZones().then((resp: any) => {
+        if (resp.status === 200 && !hasError(resp)) {
+          this.timeZones = resp.data.filter((timeZone: any) => {
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:121]: Handle the case when the status is 200, but resp does not contain data property in it.

File: src/views/TimezoneModal.vue [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        }
+      });
+    },
+    selectSearchBarText(event: any) {
+      event.target.getInputElement().then((element: any) => {
+        element.select();
+      });
+    },
+    async setUserTimeZone() {
+      return this.store
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue]:
```suggestion
      return this.store.dispatch("user/setUserTimeZone", {
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Mayank909) [reply package.json]: Ok Sir I will revert the changes.

File: src/views/TimezoneModal.vue (hunk 5 above)
Comment (Mayank909) [reply src/views/TimezoneModal.vue:121]: Sir, I found data property in the resp.

File: src/locales/en.json [hunk 7]
```diff
@@ -136,8 +141,10 @@
   "Shops": "Shops",
   "Search purchase orders and products": "Search purchase orders and products",
   "Search shipments and products": "Search shipments and products",
+  "Search time zones": "Search time zones",
   "Search users": "Search users",
   "Settings": "Settings",
+  "Select time zone" : "Select time zone",
```
Comment (meet-aniket) [inline src/locales/en.json:152]: Arrange this labels alphabetically.

File: src/views/TimezoneModal.vue (hunk 6 above)
Comment (Mayank909) [reply src/views/TimezoneModal.vue]: Done Sir

File: src/views/TimezoneModal.vue (hunk 4 above)
Comment (Mayank909) [reply src/views/TimezoneModal.vue]: Done Sir

File: src/views/TimezoneModal.vue (hunk 3 above)
Comment (Mayank909) [reply src/views/TimezoneModal.vue]: Done Sir

File: src/views/TimezoneModal.vue (hunk 2 above)
Comment (Mayank909) [reply src/views/TimezoneModal.vue]: Done  Sir

File: src/views/TimezoneModal.vue (hunk 5 above)
Comment (Mayank909) [reply src/views/TimezoneModal.vue:121]: Done Sir

**Review by meet-aniket — APPROVED** [review]

File: src/locales/en.json (hunk 7 above)
Comment (Mayank909) [reply src/locales/en.json:152]: OK sir

**Comment by Dhiraj1405** [comment]
Not priority for now. Hence, closing this pull request.
